### PR TITLE
Release SMAC3 version 0.4

### PR DESCRIPTION
**Pull request by mfeurer**

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -1,110 +1,139 @@
+# 0.4
+
+* ADD #204: SMAC now always saves runhistory files as `runhistory.py`.
```
Comment (KEggensperger) [inline changelog.md]: It is runhistory.json

File: changelog.md [hunk 2]
```diff
@@ -1,110 +1,139 @@
...
+* MAINT #205: the SMAC repository now uses codecov.io instead of coveralls.io.
+* ADD #83: support of ACLIB 2.0 parameter configuration space file.
+* FIX #206: instances are now explicitly cast to `str`, instead the instances
```
Comment (KEggensperger) [inline changelog.md]: What does "instead the instances are `None`. You mean: "if not instances are given, a single instance called 'None' is used"

File: changelog.md [hunk 3]
```diff
@@ -1,110 +1,139 @@
...
+  are `None`.
+* ADD #200: new convenience function to retrieve an `X`, `y` representation
+  of the data to feed it to a new fANOVA implementation.
+* MAINT #198: improved pSMAC interface.
+* FIX #201: improved handling of boolean arguments to SMAC.
+* FIX #194: fixes adaptive capping with re-occurring configurations.
+* ADD #190: new argument intensification percentage.
```
Comment (KEggensperger) [inline changelog.md]: The argument is called 'intensification_percentage'

File: changelog.md [hunk 4]
```diff
@@ -1,110 +1,139 @@
...
+* ADD #187: better dependency injection into main SMAC class to avoid
+  ill-configured SMAC objects.
+* ADD #161: log scenario object as a file.
+* ADD #186: show the difference between old and new incumbent in case of an
+  incumbent change.
+* MAINT #159: consistent naming of loggers.
+* ADD #128: new helper method to get the target algorithm evaluator.
+* FIX #165.
```
Comment (KEggensperger) [inline changelog.md]: default value for par = 1

**Comment by mfeurer** [comment]
@KEggensperger Thanks. My latests push addresses all requested changes.

**Comment by mfeurer** [comment]
@mlindauer @KEggensperger could you please have a look at my three latest commit?

**Pull request merged**